sympy/core/numbers.py

```diff
@@ -1173,6 +1204,321 @@ def __format__(self, format_spec):
+
+    is_complex = True
+
+    is_ComplexFloat = True
+
+    def __new__(cls, real, imag=None, prec=None):
+        if imag is None:
+            if isinstance(real, (complex, mpmath.mpc)):
+                real, imag = real.real, real.imag
+            else:
+                real, imag = real.as_real_imag()
+        if prec is None:
+            if isinstance(real, Float) and isinstance(imag, Float):
+                # no checks done; assume precisions are as desired by user
+                True
+            elif isinstance(real, Float):
+                prec = real._prec
+                imag = Float(imag, prec_to_dps(prec))
+            elif isinstance(imag, Float):
+                prec = imag._prec
+                real = Float(real, prec_to_dps(prec))
+            else:
+                real = Float(real)
+                imag = Float(imag)
+        else:
+            real = Float(real, prec)
+            imag = Float(imag, prec)
+        if real is S.NaN or imag is S.NaN:
+            return S.NaN
+        obj = Expr.__new__(cls)
+        obj.real = real
+        obj.imag = imag
+        return obj
+
+    @classmethod
+    def _new(cls, _mpc_, _prec):
+        re, im = _mpc_
+        if (re, im) == (_mpf_zero, _mpf_zero):
+            return S.Zero  # XXX this is different from ComplexFloat()
+        obj = Expr.__new__(cls)
+        # TODO: be careful, gives S.Zero
+        obj.real = Float._new(re, _prec)
+        obj.imag = Float._new(im, _prec)
+        if obj.real is S.Zero:
+            obj.real = Float(0);
+            obj.real._prec = _prec
+        if obj.imag is S.Zero:
+            obj.imag = Float(0);
+            obj.imag._prec = _prec
+        return obj
+
+    @property
+    def _mpc_(self):
+        return (self.real._mpf_, self.imag._mpf_)
+
+    @property
+    def _prec(self):
+        return min(self.real._prec, self.imag._prec)
+
+    def _hashable_content(self):
+        return (self.real, self.imag)
+
+    def __hash__(self):
+        return super(ComplexFloat, self).__hash__()
+
+    @cacheit
+    def sort_key(self, order=None):
+        # TODO: is this correct format?
+        return self.class_key(), (0, ()), (), self.real, self.imag
+
+    @property
+    def is_real(self):
+        # TODO: see is_rational above
+        #return False if self.imag.is_nonzero else None
+        return self.imag.is_zero
+
+    @property
+    def is_imaginary(self):
+        #return False if self.real.is_nonzero else None
+        return self.real.is_zero and self.imag.is_nonzero
+
+    def _eval_is_zero(self):
+        return self.real.is_zero and self.imag.is_zero
+
+    def as_real_imag(self, deep=True, **hints):
+        return (self.real, self.imag)
+
+    def __complex__(self):
+        return complex(float(self.real), float(self.imag))
+
+    def __float__(self):
+        raise TypeError("can't convert ComplexFloat to float")
+
+    def __int__(self):
+        raise TypeError("can't convert ComplexFloat to int")
```

Comment:
re: real ComplexFloat returning Float: not sure yet, in arithmetic calculations maybe.

Anyway, in native Python `int(1+0j)` gives `TypeError: can't convert complex to int` and I think `__int__` is about conversion to native Python int.  Let me know if you think I should do this anyway.